pre_commit/repository.py

```diff
@@ -146,21 +148,37 @@ def _cloned_repository_hooks(
 ) -> Tuple[Hook, ...]:
     repo, rev = repo_config['repo'], repo_config['rev']
     manifest_path = os.path.join(store.clone(repo, rev), C.MANIFEST_FILE)
-    by_id = {hook['id']: hook for hook in load_manifest(manifest_path)}
-
-    for hook in repo_config['hooks']:
-        if hook['id'] not in by_id:
-            logger.error(
-                f'`{hook["id"]}` is not present in repository {repo}.  '
-                f'Typo? Perhaps it is introduced in a newer version?  '
-                f'Often `pre-commit autoupdate` fixes this.',
-            )
-            exit(1)
+    if os.path.isfile(manifest_path):
```

Comment:
This is currently all or nothing: if target repo has a manifest, we require all configured hooks to be found in it. I think it could be better so that we use whatever is there, but allow additional local-config-only ones (i.e. if a hook is not found in manifest, use local-only config for it).

Reply:
Also, we could make all of this new stuff optional on per repo basis if this does not seem like something we'd also want to unconditionally everywhere, e.g. add a repo level `allow_local_only` or the like boolean option to trigger the new behavior with default being whatever we think is best.